docs/Execution.md

```diff
@@ -109,3 +109,75 @@ graph TD
         end
     end
 ```
+
+## Custom Execution
+
+MsQuic also supports scenarios where the application layer creates the threads that MsQuic uses to execute on.
+In this mode, the application creates one or more execution contexts that MsQuic will use to run all its internal logic.
+The application is responsible for calling down into MsQuic to allow these execution contexts to run.
+
+To create an execution context, the app much first create an event queue object, which is a platform specific type:
+
+- Windows: IOCP
```

Comment:
why do we expose a platform specific queue object rather than having a more abstracted interface?

Reply:
Such as what? The reason we use the platform queue object is because that is also a requirement for other (i.e. storage) IO. So, it's not expected to be a _new_ requirement. Also, this is all opt-in. They can continue to use MsQuic without doing this, and it will simply create the background threads still.

Reply:
No, they shouldn't create anything specific for MsQuic. If they have an epoll queue (on Linux) for their storage IO, they should reuse that for MsQuic.